File: sdk/samples/sample-app-storage/src/main/res/LoadingStateItemViewHolder.java
package com.anuchandy.learn.msal;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.azure.android.core.util.paging.PageLoadState;

public class LoadingStateItemViewHolder extends RecyclerView.ViewHolder {
    private final ProgressBar progressBar;
    private final Button retryButton;
    private final TextView errorMsg;

    private LoadingStateItemViewHolder(@NonNull View itemView, @NonNull Runnable retryRunnable) {
        super(itemView);
        this.progressBar = itemView.findViewById(R.id.progress_bar);
        this.retryButton = itemView.findViewById(R.id.retry_button);
        this.errorMsg = itemView.findViewById(R.id.error_msg);
        this.retryButton.setOnClickListener(v -> retryRunnable.run());
    }

    public static LoadingStateItemViewHolder create(ViewGroup parent, Runnable retryRunnable) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.load_state_item, parent, false);
        return new LoadingStateItemViewHolder(view, retryRunnable);
    }

    public void bind(PageLoadState loadState) {
        if (loadState != null) {
            if (loadState == PageLoadState.LOADING) {
                this.progressBar.setVisibility(View.VISIBLE);
            } else {
                this.progressBar.setVisibility(View.GONE);
            }
            if (loadState == PageLoadState.FAILED) {
                this.retryButton.setVisibility(View.VISIBLE);
            } else {
                this.retryButton.setVisibility(View.GONE);
            }
            if (false /* there is error*/) {
                this.errorMsg.setText("error-message");
            }
        }
    }
}

File: samples/sample-app-storage-kotlin/src/main/java/com/azure/android/storage/sample/kotlin/LoadingStateItemViewHolder.kt
package com.azure.android.storage.sample.kotlin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.azure.android.storage.sample.kotlin.core.util.paging.PageLoadState

class LoadingStateItemViewHolder private constructor(itemView: View, retryRunnable: Runnable) : RecyclerView.ViewHolder(itemView) {
    private val progressBar: ProgressBar = itemView.findViewById(R.id.progress_bar)
    private val retryButton: Button = itemView.findViewById(R.id.retry_button)
    private val errorMsg: TextView = itemView.findViewById(R.id.error_message)

    fun bind(loadState: PageLoadState?) {
        if (loadState != null) {
            if (loadState == PageLoadState.LOADING) {
                progressBar.visibility = View.VISIBLE
            } else {
                progressBar.visibility = View.GONE
            }
            if (loadState == PageLoadState.FAILED) {
                retryButton.visibility = View.VISIBLE
            } else {
                retryButton.visibility = View.GONE
            }
            if (false /* there is error*/) {
                errorMsg.text = "error-message"
            }
        }
    }

    companion object {
        fun create(parent: ViewGroup, retryRunnable: Runnable): LoadingStateItemViewHolder {
            val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.load_state_item, parent, false)
            return LoadingStateItemViewHolder(view, retryRunnable)
        }
    }

    init {
        retryButton.setOnClickListener { retryRunnable.run() }
    }
}
